R/addFilesBySource.R

```diff
@@ -0,0 +1,81 @@
+#' @param connection the connection to data.world
+#' @param dataset the "agentid/datasetid" for the dataset against which to execute the query
+#' @param name the filename including the file extension. If a file by that name already exists in the dataset, the file will be updated/overwritten.
+#' @param url the public, full direct-download path to file
+#' @examples
+#' connection <- data.world(token = "YOUR_API_TOKEN_HERE")
+#' data.world::addFileBySource(connection, dataset = "ownerid/datasetid", name = "file.csv", url = "https://data.world/some_file.csv")
+#' @export
+addFileBySource <- function(connection, dataset, name, url) {
+  UseMethod("addFileBySource")
+}
+
+#' @export
+addFileBySource.default <- function(connection, dataset, name, url) {
+  print("nope.")
+}
+
+#' @export
+addFileBySource.data.world <- function(connection, dataset, name, url) {
+  fileBatchUpdateRequest <- data.world::FileBatchUpdateRequest()
+  fileBatchUpdateRequest <- data.world::addFile(request = fileBatchUpdateRequest, name = name, url = url)
+  ret <- data.world::addFilesBySource(connection = connection, dataset = dataset, request = fileBatchUpdateRequest)
+  ret
+}
+
+#' publish file(s) on the web to be added to a data.world dataset via `data.world::FileBatchUpdateRequest`
+#' @param connection the connection to data.world
+#' @param dataset the "agentid/datasetid" for the dataset against which to execute the query
+#' @param request a requet of type `data.world::FileBatchUpdateRequest`
+#' @examples
+#' request <- data.world::FileBatchUpdateRequest()
+#' request <- data.world::addFile(request = request, name = "file.csv", url = "https://data.world/some_file.csv")
+#' connection <- data.world(token = "YOUR_API_TOKEN_HERE")
+#' data.world::addFilesBySource(connection, dataset = "ownerid/datasetid", request = request)
+#' @export
+addFilesBySource <- function(connection, dataset, request) {
+  UseMethod("addFilesBySource")
+}
+
+#' @export
+addFilesBySource.default <- function(connection, dataset, request) {
+  print("nope.")
+}
+
+#' @export
+addFilesBySource.data.world <- function(connection, dataset, request) {
+  apiUrl = sprintf("%sdatasets/%s/files", connection$baseDWApiUrl, dataset)
+  contentTypeHeader <- "application/json"
+  authHeader = sprintf("Bearer %s", connection$token)
+  response <- httr::POST( apiUrl,
+                          body = rjson::toJSON(request),
+                          httr::add_headers(
+                            "Content-Type" = contentTypeHeader,
+                            "Authorization" = authHeader),
+                          httr::progress(),
+                          httr::user_agent(data.world::userAgent()))
+  ret <- httr::http_status(response)
+  if (response$status_code == 200 | response$status_code == 202) {
+    ret <- data.world::SuccessMessage(rjson::fromJSON(httr::content(x=response, as='text')))
+  } else {
+    ret <- data.world::ErrorMessage(rjson::fromJSON(httr::content(x=response, as='text')))
```

Comment:
Consider converting `ErrorMessage` into proper R errors. Same comment applies to all endpoints.